src/main/java/crystal/Parser.java

```diff
@@ -0,0 +1,75 @@
+package crystal;
+
+import crystal.command.Command;
+import crystal.command.ListCommand;
+import crystal.command.UnmarkCommand;
+import crystal.command.MarkCommand;
+import crystal.command.TodoCommand;
+import crystal.command.DeadlineCommand;
+import crystal.command.EventCommand;
+import crystal.command.ByeCommand;
+import crystal.command.DeleteCommand;
+import crystal.command.FindCommand;
+
+/**
+ * Represents the Parser task.
+ *
+ */
+
+public class Parser {
+
+
+    /**
+     * Returns the respective command object based on the command
+     * the user enters.
+     *
+     * @param command User command
+     * @throws CrystalException When user command is not recognised
+     */
+    public static Command parse(String command) throws CrystalException {
+        if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.contains("unmark")) {
+            String getNum = command.replaceAll("[^0-9]", "");
+            int num = Integer.parseInt(getNum);
+            return new UnmarkCommand(num);
+        } else if (command.contains("mark")) {
+            String getNum = command.replaceAll("[^0-9]", "");
+            int num = Integer.parseInt(getNum);
+            return new MarkCommand(num);
+        } else if (command.contains("todo")) {
+            String stringNoTodo = command.replace("todo", "");
+            if (stringNoTodo.length() == 0) {
+                throw new CrystalException("☹ OOPS!!! The description of a todo cannot be empty.");
+            }
+            return new TodoCommand(stringNoTodo);
+        } else if (command.contains("deadline")) {
+            String stringNoDeadline = command.replace("deadline", "");
+            int index = stringNoDeadline.lastIndexOf("/by");
+            String description = stringNoDeadline.replace(stringNoDeadline.substring(index), ""); //description of task
+            stringNoDeadline = stringNoDeadline.substring(index + 3); //date/time for by portion
+            return new DeadlineCommand(description, stringNoDeadline);
+        } else if (command.contains("event")) {
+            String stringNoEvent = command.replace("event", "");
```

Comment:
All these functionality in the commands block can maybe be abstracted into functions. Good job on cleaning your input string , looks like a good defensive programming practice